obspy/imaging/spectrum.py

```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*-
+
+"""
+Plotting the spectrum of seismograms
+"""
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+from future.builtins import *  # NOQA @UnusedWildImport
+import numpy as np
+import matplotlib.pyplot as plt
+from scipy import fft
+
+def plot_spectrum(data, samp_rate, outfile=None, fmt=None,
+             axes=None, title=None, show=True):
+
+    """
+     Compute and plot the spectrum of the input data
+     """
+
+    # enforce float for samp_rate
+    samp_rate = float(samp_rate)
+
+    npts = len(data)
+
+    # Compute the FFT
+    npts = len(data)
+    k = np.arange(npts)
+    T = npts/samp_rate
+    frq = k/T  # two sides frequency range
+    frq = frq[range(npts/2)]  # one side frequency range
+
+    X = fft(data)/npts  # fft computing and normalization
```

Comment:
No that cannot work. The spectrum still has to be complex. If you really want to work with complex values you can add zeros after the first sample (and after the last if n is odd) and set the dtype to a complex array...

```python
In [3]: temp = scipy.fftpack.rfft(np.arange(10))

In [4]: i = temp[0]

In [5]: temp[0] = i

In [6]: rfft = np.concatenate([[i], temp, [0]])

In [7]: rfft.dtype = np.complex128

In [7]: rfft
Out[7]:
array([ 45.+45.j        ,  -5.+15.38841769j,  -5. +6.8819096j ,
        -5. +3.63271264j,  -5. +1.62459848j,  -5. +0.j        ])

In [8]: scipy.fftpack.fft(np.arange(10))
Out[8]:
array([ 45. +0.j        ,  -5.+15.38841769j,  -5. +6.8819096j ,
        -5. +3.63271264j,  -5. +1.62459848j,  -5. +0.j        ,
        -5. -1.62459848j,  -5. -3.63271264j,  -5. -6.8819096j ,
        -5.-15.38841769j])
```